lecture_2/homework/app/controllers/clans/warriors_controller.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Clans
+  class WarriorsController < ApplicationController
+    def show
+      render json: warrior, include: '*.'
+    end
+
+    def index
+      warriors = if building
+                   building.warriors
+                 else
+                   clan.warriors
+                 end
+
+      if params.key?(:alive)
+        if params[:alive].to_i == 0
+          render json: warriors.dead
+        else
+          render json: warriors.alive
+        end
+      else
+        render json: warriors
+      end
+    end
+
+    def create
+      warrior = if building
+                  building.warriors.create!(warrior_params)
+                else
+                  clan.warriors.create!(warrior_params)
+                end
+
+      render json: warrior, status: 201
+    end
+
+    def update
+      warrior.update!(warrior_params)
+
+      render json: warrior
+    end
+
+    def destroy
+      warrior.destroy!
+    end
+
+    private
+
+    def clan
+      @clan ||= Clan.find(params[:clan_id])
+    end
+
+    def warrior
+      @warrior ||= Warrior.find_by!(id: params[:id])
+    end
+
+    def weapon
+      @weapon ||= warrior.weapon
+    end
+
+    def building
+      if params.key?(:castle_id)
```

Comment:
Widzę, że nie użyłeś STI, bo Twoje budynki mają sporo dodatkowych atrybutów. Żeby uniknąć takiej konstrukcji ifów można przesyłać typu budowli w parametrach i na tej podstawie wybierać klasę, której będziemy szukać (musimy tylko pamietać, żeby zwalidować czy wybrana klasa jest jedną z dozwolonych). Innym rozwiązaniem może być zrobienie dodatkowej tabeli która będzie działać jak "lista" wszystkich budynków. Wtedy zamiast na przykład podawać id zamku to podajemy jego id na tej dodatkowej liście - ponieważ mamy tam wszystkie budynki to na pewno go znajdziemy.